### PR TITLE
Initial framework api for read/write 07/13/2022

### PR DESCRIPTION
**Pull request by dman776**

### CODE REVIEW

File: files/usr/lib/lua/aredn/controller/setup_basic.lua [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  radio = aredn_info.getMeshRadioDevice()
+  data['meshrf']['enabled'] = aredn_info.isMeshRadioEnabled(radio)
+  data['meshrf']['ip'] = aredn_info.getInterfaceIPAddress("wifi")
+  data['meshrf']['netmask'] = aredn_info.getInterfaceNetmask("wifi")
+  data['meshrf']['distance'] = aredn_info.getMeshRadioDistance(radio)
+  data['meshrf']['bw'] = aredn_info.getChannelBW(radio)
+  data['meshrf']['channel'] = aredn_info.getChannel(radio)
+  data['meshrf']['power'] = aredn_info.getTXPower()
+  data['meshrf']['maxpower'] = aredn_hardware.wifi_maxpower(data['meshrf']['channel'])
+
```
Comment (aanon4) [inline files/usr/lib/lua/aredn/controller/setup_basic.lua:88]: How about:

local ssid =  aredn_info.getSSID()
local radio = aredn_info.getMeshRadioDevice()
data.meshrf = {
    ssid_full = ssid,
    ssid_prefix =  ssid:split("-")[1],
    enabled = aredn_info.isMeshRadioEnabled(radio),
    ip = aredn_info.getInterfaceIPAddress("wifi"),
   ... etc ...
}

I think it's a little easier to read (and probably more efficient to construct)

Comment (aanon4) [inline files/usr/lib/lua/aredn/controller/setup_basic.lua:88]: How about:

local ssid =  aredn_info.getSSID()
local radio = aredn_info.getMeshRadioDevice()
data.meshrf = {
    ssid_full = ssid,
    ssid_prefix =  ssid:split("-")[1],
    enabled = aredn_info.isMeshRadioEnabled(radio),
    ip = aredn_info.getInterfaceIPAddress("wifi"),
   ... etc ...
}

I think it's a little easier to read (and probably more efficient to construct)

Comment (dman776) [reply files/usr/lib/lua/aredn/controller/setup_basic.lua]: do you just mean change to dot notation, like this commit?

Comment (aanon4) [reply files/usr/lib/lua/aredn/controller/setup_basic.lua]: I prefer the dot notation, but I think you can go one step further like in my example where you unwrap the multiple assignments. But it's not a religious conviction so go with what you think is easier to understand.

**Comment by dman776** [comment]
thoughts on merging this to allow for easier newui development for r/w pages?

**Review by aanon4 — APPROVED** [review]
For sure

**Pull request merged**